submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/service/StatsMongoDataService.kt

```diff
@@ -0,0 +1,66 @@
+package ac.uk.ebi.biostd.persistence.doc.service
+
+import ac.uk.ebi.biostd.persistence.common.exception.StatNotFoundException
+import ac.uk.ebi.biostd.persistence.common.model.SubmissionStat
+import ac.uk.ebi.biostd.persistence.common.model.SubmissionStatType
+import ac.uk.ebi.biostd.persistence.common.request.PaginationFilter
+import ac.uk.ebi.biostd.persistence.common.service.StatsDataService
+import ac.uk.ebi.biostd.persistence.doc.db.data.SubmissionDocDataRepository
+import ac.uk.ebi.biostd.persistence.doc.model.DocStat
+import ac.uk.ebi.biostd.persistence.doc.model.DocSubmission
+import ac.uk.ebi.biostd.persistence.doc.model.SingleSubmissionStat
+import org.springframework.data.domain.PageRequest
+
+class StatsMongoDataService(
+    private val subDataRepository: SubmissionDocDataRepository
+) : StatsDataService {
+    override fun findByType(submissionStatType: SubmissionStatType, filter: PaginationFilter): List<SubmissionStat> =
+        subDataRepository
+            .findAllByStatType(submissionStatType, PageRequest.of(filter.pageNumber, filter.limit))
+            .content
+            .map { toSubmissionStat(it, submissionStatType) }
+
+    override fun findByAccNoAndType(accNo: String, submissionStatType: SubmissionStatType): SubmissionStat =
+        subDataRepository
+            .findByAccNoAndStatType(accNo, submissionStatType)
+            ?.let { toSubmissionStat(it, submissionStatType) }
+            ?: throw StatNotFoundException(accNo, submissionStatType)
+
+    override fun saveAll(stats: List<SubmissionStat>): List<SubmissionStat> =
+        stats.filterValid()
```

Comment:
not following why we need to filter valid this looks to me like and accept a request and ignore errors, I think the correct thing to do is that an API fail if input is invalid.

Reply:
We already discussed this with the frontend guys when this was developed and we agreed the API shouldn't fail in this case. Should we really change this behavior?

Reply:
lets just reference TODO ticket to remove this hack and allow stats for expired versions